languages/rust/exercises/concept/structs/.docs/instructions.md

```diff
@@ -0,0 +1,43 @@
+It is often useful to group a collection of items together, and handle those groups as units. In Rust, we call such a group a struct, and each item one of the struct's fields. A struct defines the general set of fields available, but a particular example of a struct is called an instance.
+
+Furthermore, structs can have methods defined on them, which have access to the fields. The struct itself in that case is referred to as `self`. When a method uses `&mut self`, the fields can be changed, or mutated. When a method uses `&self`, the fields cannot be changed: they are immutable. Controlling mutability helps the borrow-checker ensure that entire classes of concurrency bug just don't happen in Rust.
+
+In this exercise, you'll be implementing two kinds of methods on a struct. The first are generally known as getters: they expose the struct's fields to the world, without letting anyone else mutate that value. In Rust, these methods idiomatically share the name of the field they expose, i.e., if we have a getter method that fetches a struct field called `name`, the method would simply be called `name()`.
+
+You'll also be implementing methods of another type, generally known as setters. These change the value of the field. Setters aren't very common in Rust--if a field can be freely modified, it is more common to just make it public--but they're useful if updating the field should have side effects, or for access control: a setter marked as `pub(crate)` allows other modules within the same crate to update a private field, which can't be affected by the outside world.
```

Comment:
I would also argue that there still isn't a story/theme to this exercise. Maybe I wasn't clear on what a story/theme should be, but it shouldn't be a technical description of what the student should do. Instead, it should be a (fictional) background that places the exercise in some story-like context. Here are some examples:

- The [C# `floating-point-numbers` exercise](https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/floating-point-numbers/.docs/instructions.md) uses a savings account story/theme for its tasks.
- The [Elixir `booleans` exercise](https://github.com/exercism/v3/blob/master/languages/elixir/exercises/concept/booleans/.docs/instructions.md) uses Pac-Man as its story/theme.
- The [JavaScript `arrays` exercise](https://github.com/exercism/v3/blob/master/languages/javascript/exercises/concept/arrays/.docs/instructions.md) uses a magician with a deck of cards as its story/theme.

This exercise can either re-use an existing theme (we have a [complete list of stories here](https://github.com/exercism/v3/blob/master/reference/stories/README.md)), or come up with one. Whatever you prefer.

The tasks in this file also don't follow the [docs](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md#docsinstructionsmd). Each task should have a separate header (`### 2. Do Y`), which you can find examples of in the above-mentioned examples.

Reply:
Story, compatible with existing work:

> You are implementing a health-monitoring service, and you need to keep track of your users' basic statistics.

I'm surprised you're attaching much importance to the story; possibly the times have changed, but what I recall was that it was agreed that concept exercises in particular should be very brief, and that stories were really better for practice exercises. The _point_ of a concept exercise, as I understand it, is to pose a very small obstacle to someone who already knows the concept, and a gentle introduction for someone who is encountering it for the first time. Given that they're meant to be numerous and bite-sized, I thought we'd agreed that there was no point in fleshing out a story.

Reply:
> You are implementing a health-monitoring service, and you need to keep track of your users' basic statistics.

Yep, this would totally work. A story/theme can be very simple, with the basics exercises all using a very simple, cooking-related theme (see the [C# basics exercise](https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/basics/.docs/instructions.md)).

> I'm surprised you're attaching much importance to the story; possibly the times have changed, but what I recall was that it was agreed that concept exercises in particular should be very brief, and that stories were really better for practice exercises. 

We want to have a story as we've found that students respond better to working on an exercise when they have some context they can place it in. We do want Concept Exercises to be brief, but that doesn't mean they can't have a story. Concept Exercises should be brief due to their implementation being simpler, but they should have a story. The story for Practice Exercises can be both simpler or more complex, there is no general rule here I feel, as it depends so much on what the Practice Exercise is about.

> The point of a concept exercise, as I understand it, is to pose a very small obstacle to someone who already knows the concept, and a gentle introduction for someone who is encountering it for the first time.

But this does not rule out having a story/theme, does it? 

> I thought we'd agreed that there was no point in fleshing out a story.

I don't think anyone from the Exercism team has ever said this :)